### DISCUSSION TITLE
How to use " button-actions-execute query"?

### DISCUSSION THREAD

**Discussion by lipan1024** [Budibase/budibase#2628] [Help] [answered]
I want to query SQL based on time conditions,this is my design:
![image](https://user-images.githubusercontent.com/86212306/133709623-b590f5c9-5c53-43fa-8697-8fa20bcd747f.png)

I defined an action "execute query" on the button,it seems to work as i designed,but i don't know how to get the data into my table:
![image](https://user-images.githubusercontent.com/86212306/133710327-9487bf24-a81a-4434-94e0-18e45c870fd0.png)

   help...

**Comment by mjashanks** [comment, marked as answer]
Hi @lipan1024 

Have you added a SQL datasource in the Data section? https://docs.budibase.com/building-apps/data/tables/postgresql-mysql-and-sql-server?

On your Data Provider, the `Data` setting is set to `New Form`. You will need to set this to one of your SQL tables, or Custom Queries, if that's what you are using. If you are using a Postgres or MySQL connection, all your tables should appear in this List.

Then, if you click the "Define Filters" button, you should be able to use values from your data inputs to pass into the filter.

You shouldn't need your "Search" button. The data source will automatically refresh when the date is changed.

**Reply by lipan1024** [reply]
Thanks, yes, it can be done by using filters, but I still don't know what scenario the " button-actions-execute query" is used in

**Reply by mjashanks** [reply]
Execute Query is used for custom queries. i.e. E.g. a custom SQL query (not the in-built ones you get when you "fetch tables"). Other connectors like RestAPI and Non-SQL connectors also use custom queries.

**Reply by lipan1024** [reply]
thanks, So... can I use the data set returned by the custom queries in my pages? What should I do if I can

**Reply by mjashanks** [reply]
In this app, I have added a postgres connector. See the "Add Query" button in the bottom right? This is where you would create your query.

Once you have done that, then
1. Click on your Data Provider (in your screen)
2. Set the `Data` property to your newly added query.

![image](https://user-images.githubusercontent.com/3524181/133752278-db89e76a-a23c-4d45-aaf0-e498442164dc.png)
